src/MOI_wrapper/MOI_wrapper.jl

```diff
@@ -776,46 +771,51 @@ function MOI.get(model::Optimizer, ::MOI.RelativeGap)
     return isnan(gap) ? Inf : gap
 end
 
-function MOI.get(
-    model::Optimizer,
-    attr::MOI.VariablePrimal,
-    x::MOI.VariableIndex,
-)
-    MOI.check_result_index_bounds(model, attr)
-    return _unsafe_wrap_cbc_array(
-        model,
-        Cbc_getColSolution,
-        Cbc_getNumCols(model),
-        x.value,
-    )
+_update_cache(::Optimizer, data::Vector{Float64}, ::Any, ::Any) = data
+
+function _update_cache(model::Optimizer, ::Nothing, f_p::F, f_n::G) where {F,G}
+    p = f_p(model)
```

Comment:
Couldn't you skip this if `variable_primal` is not `nothing` ? And then you reset it to `nothing` when `optimize!` is called.

Reply:
See line 774, we do skip it.